### PR TITLE
Add zoom param to plot

### PR DESCRIPTION
**Pull request by EricWay1024**
Allow zooming in/out without up/downsampling

### CODE REVIEW

**Comment by MichaelClerx** [comment]
1. Does the 'zoom' look the same as our upsampling? If so, then we can remove the upsampling option
2. Do we need to set the `interpolation` type on line 209 to something else when doing this? (will also affect answer to 1)

**Comment by EricWay1024** [comment]
When `interpolation` type is set to `bilinear`, zoom and upsampling do produce the same plot.

`upsampling = 4`
image

`downsampling = 1, zoom = 4, interpolation = 'bilinear'`
image

I'll remove `upsampling` then.

File: nevis/_plot.py [hunk 1]
```diff
@@ -51,13 +51,13 @@ def plot(boundaries=None, labels=None, trajectory=None, points=None,
         Show the 2-letter 2-number grid squares (10km by 10km)
     ``downsampling``
         Set to any integer to set the amount of downsampling (the ratio of data
-        points to pixels in either direction). If ``zoom`` is not set, the
-        default is 27, which creates a reasonable plot for the full GB
-        dataset; if ``zoom`` is set, the default is 1.
+        points to pixels in either direction). If ``zoom`` is less than or
+        equal to 1, the default is 27, which creates a reasonable plot for the
+        full GB dataset; if ``zoom`` is greater than 1, the default is 1.
```
Comment (MichaelClerx) [inline nevis/_plot.py:56]:
```suggestion
        points to pixels in either direction). The default setting is 27, which
        creates a reasonable plot for the full GB data set. Note that the
        ``zoom`` setting overrules this argument: if ``zoom`` is set to
        anything other than 1, then no downsampling is performed.
```

File: nevis/_plot.py [hunk 2]
```diff
@@ -51,14 +51,14 @@ def plot(boundaries=None, labels=None, trajectory=None, points=None,
...
-        to 1. ``boundaries`` must be set for upsampling to take effect. Note
-        that this might take a long time, depending on size of the boundaries.
+        points to pixels in either direction). The default setting is 27, which
+        creates a reasonable plot for the full GB data set. Note that the
+        ``zoom`` setting overrules this argument: if ``zoom`` is set to
+        anything other than 1, then no downsampling is performed.
+    ``zoom``
+        Set to any positive number. If greater than 1, the map is zoomed in by
+        this factor; if less than 1, the map is zoomed out by this factor.
+        ``zoom`` takes effect on top of downsampling. Default is 1.
```
Comment (MichaelClerx) [inline nevis/_plot.py:61]:
```suggestion
        Set to any positive number. If greater than 1, the map is zoomed in by
        this factor; if less than 1, the map is zoomed out by this factor. No
        downsampling is performed when ``zoom != ``. The default value is 1.
```

**Comment by MichaelClerx** [comment]
What's the use case for zoom < 1 ?

**Comment by EricWay1024** [comment]
It's actually not very useful when z < 1 as we can use `downsampling`...

**Comment by EricWay1024** [comment]
How about when `zoom < 1` we automatically set `downsampling = round(1 / zoom)`?

**Comment by MichaelClerx** [comment]
> `downsampling = round(1 / zoom)`

Good idea!
Yes let's do that and remove the `downsampling` argument. Just make sure it's well explained in the docstring :D

File: examples/fitting-with-pints.ipynb [hunk 3]
```diff
@@ -337,7 +337,7 @@
     "    labels=labels,\n",
     "    trajectory=trajectory,\n",
     "    points=points,\n",
-    "    downsampling=27,\n",
+    "    zoom=0.03,\n",
```
Comment (MichaelClerx) [inline examples/fitting-with-pints.ipynb:340]:
```suggestion
    "    zoom=1 / 27,\n",
```

Comment (MichaelClerx) [reply examples/fitting-with-pints.ipynb:340]: Would it make sense to specify the default zoom value like this?

Comment (EricWay1024) [reply examples/fitting-with-pints.ipynb:340]: I suppose so!

File: nevis/__init__.py [hunk 4]
```diff
@@ -91,6 +91,6 @@ def write_test_figure(path='gb-small.png'):
     """
     gb()
     labels = {'Ben Nevis': ben(), 'Holme Fen': fen()}
-    fig, ax, heights, g = plot(labels=labels, downsampling=32)
+    fig, ax, heights, g = plot(labels=labels)
```
Comment (MichaelClerx) [inline nevis/__init__.py:94]:
```suggestion
    fig, ax, heights, g = plot(labels=labels, zoom=1 / 32)
```

File: nevis/_plot.py [hunk 5]
```diff
@@ -49,16 +49,11 @@ def plot(boundaries=None, labels=None, trajectory=None, points=None,
...
-        Set to any integer to set the amount of upsampling (the ratio of
-        upsampled data points to original ones in either direction) using
-        linear interpolation. When not None, ``downsampling`` is overwritten
-        to 1. ``boundaries`` must be set for upsampling to take effect. Note
-        that this might take a long time, depending on size of the boundaries.
+    ``zoom``
+        Set to any positive number. When ``zoom`` is greater than 1, "zoom in"
+        the map (using bilinear interpolation); when ``zoom`` is less than 1,
+        "zoom out" the map (using downsampling). The default value is
+        ``1 / 27``, which creates a reasonable plot for the full GB data set.
```
Comment (MichaelClerx) [inline nevis/_plot.py:56]:
```suggestion
        Adjust plot size by interpolating (``zoom > 1``) or downsampling (
        ``zoom < 1``).
        
        To downsample by e.g. a factor 10, set ``zoom = 1 / 10``. To 
        interpolate and show e.g. 3x3 pixels per data point, set ``zoom = 3``.
        Interpolation is performed using matplotlib's "bilinear interpolation".

        The default value is ``1 / 27``, which creates a reasonably sized plot
        for the full GB data set.
```

File: nevis/_plot.py [hunk 6]
```diff
@@ -80,16 +75,12 @@ def plot(boundaries=None, labels=None, trajectory=None, points=None,
...
-                f'Upsampling with factor {upsampling}, '
-                'downsampling is set to 1'
-            )
-        if boundaries is None:
-            print('Warning: Upsampling requires boundaries to be set.')
+    # Calculate downsampling using zoom
+    if zoom > 1:
         downsampling = 1
+    else:
+        downsampling = round(1 / zoom)
```
Comment (MichaelClerx) [inline nevis/_plot.py:82]:
```suggestion
        downsampling = int(round(1 / zoom))
```

Comment (EricWay1024) [reply nevis/_plot.py:82]: `round` by default returns a `int` (https://docs.python.org/3/library/functions.html#round)

**Pull request merged**